source/scriptHandler.py

```diff
@@ -341,6 +346,90 @@ def isScriptWaiting():
 	return bool(_numScriptsQueued)
 
 
+def _initEnglishGesturesConfig():
+	import os
+	import configobj
+	import globalVars
+
+	filename = os.path.join(globalVars.appDir, "locale", "en", "gestures.ini")
+	gestureConfig = configobj.ConfigObj(infile=filename, encoding="UTF-8", indent_type="\t")
+	return gestureConfig
+
+
+_FindScriptResultsT = list[tuple["inputCore.GlobalGestureMap", list["inputCore.InputGestureScriptT"]]]
+
+
+def _findScriptInGlobalMap(gesture: str) -> _FindScriptResultsT:
+	"""Find global scripts for a gesture"""
+	from inputCore import manager, normalizeGestureIdentifier
+
+	assert manager
+	normalizedGesture = normalizeGestureIdentifier(gesture)
+	globalScripts = [manager.userGestureMap, manager.localeGestureMap, manager.defaultGestureMap]
+	conflicts: _FindScriptResultsT = []
+	for scriptMap in globalScripts:
+		if normalizedGesture in scriptMap._map:
+			conflicts.append((scriptMap, scriptMap._map[normalizedGesture]))
+	return conflicts
+
+
+def _checkForConflicts(script: types.FunctionType, gestures: list[str]):
+	"""
+	Check for conflicts in the global script map for an add-on script registered with the @script decorator.
+	Log an error if found.
+	"""
+	if not script.__qualname__.startswith("GlobalPlugin"):
+		# Only check for conflicts in add-ons.
+		return
+
+	for gesture in gestures:
+		if foundScripts := _findScriptInGlobalMap(gesture):
+			foundScriptLocations = [
+				f'{", ".join(".".join(name) for name in mapScripts)} ({map.fileName})'
+				for map, mapScripts in foundScripts
+			]
+			log.error(
+				f"{script.__qualname__}: Gesture {gesture} conflicts with another script {foundScriptLocations}. "
+				"Consider using a different gesture. ",
+			)
+
+
+def _registerEnglishGesture(script: types.FunctionType, gestures: list[str], description: str):
+	"""
+	Register a gesture for a script registered with the @script decorator.
+	Creates an entry in the English gestures.ini file.
+	This is useful when running from source to migrate gesture definitions from script decorators to the gestures.ini file.
+	"""
+	if not NVDAState.isRunningAsSource():
+		return
+	className = script.__qualname__.replace(f".{script.__name__}", "")
+	if className == "GlobalPlugin" or script.__module__.startswith("tests."):
+		# add-ons shouldn't be registered.
+		# In future, check for conflicts when registering an add-on gesture.
+		return
+	scriptLocation = f"{script.__module__}.{className}"
+	scriptName = script.__name__.replace("script_", "")
```

Comment:
I think we only want to remove the prefix "`script_`".
```suggestion
	scriptName = script.__name__.removeprefix("script_")
```